src/cargo/core/compiler/build_context/target_info.rs

```diff
@@ -588,6 +590,10 @@ impl TargetInfo {
             .iter()
             .any(|sup| sup.as_str() == split.as_str())
     }
+
+    pub fn get_target_envs(&self) -> CargoResult<&BTreeMap<String, Option<OsString>>> {
+        return Ok(self.crate_type_process.get_envs());
+    }
```

Comment:
Why does this return a `CargoResult` if its always `Ok`.  The `unwrap` in the caller seemed dubious.

Reply:
I was wrong.